pallets/xyk/src/lib.rs

```diff
@@ -948,6 +875,220 @@ impl<T: Config> AMM<T::AccountId, AssetId, AssetPair, Balance> for Pallet<T> {
 	/// Note : the execution should not return error as everything was previously verified and validated.
 	#[transactional]
 	fn execute_buy(transfer: &AMMTransfer<T::AccountId, AssetId, AssetPair, Balance>) -> DispatchResult {
+		Self::do_execute_buy(&transfer)
+	}
+
+	fn get_min_trading_limit() -> Balance {
+		T::MinTradingLimit::get()
+	}
+
+	fn get_min_pool_liquidity() -> Balance {
+		T::MinPoolLiquidity::get()
+	}
+
+	fn get_max_in_ratio() -> u128 {
+		T::MaxInRatio::get()
+	}
+
+	fn get_max_out_ratio() -> u128 {
+		T::MaxOutRatio::get()
+	}
+
+	fn get_fee(_pool_account_id: &T::AccountId) -> (u32, u32) {
+		T::GetExchangeFee::get()
+	}
+}
+
+pub struct AllowAllPools();
+
+impl CanCreatePool<AssetId> for AllowAllPools {
+	fn can_create(_asset_a: AssetId, _asset_b: AssetId) -> bool {
+		true
+	}
+}
+
+use hydradx_traits::router::{Executor, ExecutorError, PoolType, TradeCalculation};
+
+impl<T: Config> Executor<T::AccountId, AssetId, Balance> for Pallet<T> {
+	type TradeCalculationResult = TradeCalculation<Balance>;
+	type Error = ();
+
+	fn calculate_sell(
+		pool_type: PoolType<AssetId>,
+		asset_in: AssetId,
+		asset_out: AssetId,
+		amount_in: Self::TradeCalculationResult,
+	) -> Result<Self::TradeCalculationResult, ExecutorError<Self::Error>> {
+		if pool_type != PoolType::XYK {
+			return Err(ExecutorError::NotSupported);
+		}
+
+		let pair = AssetPair { asset_in, asset_out };
```

Comment:
the whole trait implementation should be probably in a separate file and use pallet methods instead of pallet using trade execution trait

the relationship should be inverted in my opinion

Reply:
extracting to separate file is clear, good point, just done it!

but I am not sure what you mean by inverting. In the implementation of `Executor`/ (or with new name `TradeExecution`), we already calling the pallet methods, such as `get_pair_id`, `do_validate_sell_common` or `do_execute_sell`.  Or I might miss something...

Reply:
meanining that common functions should be implemented in XYK without necessity to implement some Executor trait. Executor trait then can be implemented in separate file by just calling whatever it needs to do in XYK

XYK implementation itself shouldn't be contained in executor trait - this was my original issue with @enthusiastmartin implementation that he basically reimplemented XYK for router, you moved XYK impl to router instead

 I would like to keep XYK impl in XYK

Reply:
for example i would keep `execute_sell` exactly as is and just called it from the `TradeExecution` trait impl

Reply:
in a perfect world, the whole xyk/lib would stay the same and route executor imp would be just in another file

Reply:
Totally agree with you! We have a session with Martin soon where we will try to simplify and clean the things up.